pkg/server/datastore/sqldriver/awsrds/awsrds.go

```diff
@@ -0,0 +1,168 @@
+package awsrds
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/jackc/pgx/v5"
+	"github.com/jinzhu/gorm"
+	"github.com/lib/pq"
+)
+
+const (
+	MySQLDriverName     = "aws-rds-mysql"
+	PostgresDriverName  = "aws-rds-postgres"
+	getAuthTokenTimeout = time.Second * 30
+)
+
+// nowFunc returns the current time and can overridden in tests.
+var nowFunc = time.Now
+
+// Config holds the configuration settings to be able to authenticate to a
+// database in the AWS RDS service.
+type Config struct {
+	Region          string `json:"region"`
+	AccessKeyID     string `json:"access_key_id"`
+	SecretAccessKey string `json:"secret_access_key"`
+	Endpoint        string `json:"endpoint"`
+	DbUser          string `json:"dbuser"`
+	DriverName      string `json:"driver_name"`
+	ConnString      string `json:"conn_string"`
+}
+
+// FormatDSN returns a DSN string based on the configuration.
+func (c *Config) FormatDSN() (string, error) {
+	dsn, err := json.Marshal(c)
```

Comment:
nit: this seems ok since it's wholly contained by the driver implementation. I wonder if encoded url.Values might be a little more natural choice...